From 88885e466fe282c16c1d62c0ef587400b54cd40d Mon Sep 17 00:00:00 2001
From: Mohammed Naser <mnaser@vexxhost.com>
Date: Sun, 2 Mar 2025 12:30:08 -0500
Subject: [PATCH] Add missing release namespace for manila charts

The manila charts are missing the namespace and when templating
things with Helm, it will omit it and be problematic.  The cinder
chart has the namespaces set everywhere and its better to be
explicit so this adds it to the manila chart.
---
 openstack-manila-csi/templates/controllerplugin-role.yaml    | 1 +
 .../templates/controllerplugin-rolebinding.yaml                  | 1 +
 openstack-manila-csi/templates/controllerplugin-service.yaml | 1 +
 .../templates/controllerplugin-serviceaccount.yaml               | 1 +
 .../templates/controllerplugin-statefulset.yaml                  | 1 +
 openstack-manila-csi/templates/nodeplugin-daemonset.yaml     | 1 +
 .../templates/nodeplugin-rules-clusterrole.yaml                  | 1 +
 .../openstack-manila-csi/templates/nodeplugin-serviceaccount.yaml   | 1 +
 openstack-manila-csi/templates/runtimeconfig-cm.yaml         | 1 +
 9 files changed, 9 insertions(+)

diff --git a/openstack-manila-csi/templates/controllerplugin-role.yaml b/openstack-manila-csi/templates/controllerplugin-role.yaml
index 36a67f9921..c3cff29aac 100644
--- a/openstack-manila-csi/templates/controllerplugin-role.yaml
+++ b/openstack-manila-csi/templates/controllerplugin-role.yaml
@@ -2,6 +2,7 @@ kind: Role
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: {{ include "openstack-manila-csi.controllerplugin.fullname" . }}
+  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "openstack-manila-csi.controllerplugin.labels" .  | nindent 4 }}
 rules:
diff --git a/openstack-manila-csi/templates/controllerplugin-rolebinding.yaml b/openstack-manila-csi/templates/controllerplugin-rolebinding.yaml
index 31f9fb2610..f34d6062dd 100644
--- a/openstack-manila-csi/templates/controllerplugin-rolebinding.yaml
+++ b/openstack-manila-csi/templates/controllerplugin-rolebinding.yaml
@@ -2,6 +2,7 @@ kind: RoleBinding
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: {{ include "openstack-manila-csi.controllerplugin.fullname" . }}
+  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "openstack-manila-csi.controllerplugin.labels" .  | nindent 4 }}
 subjects:
diff --git a/openstack-manila-csi/templates/controllerplugin-service.yaml b/openstack-manila-csi/templates/controllerplugin-service.yaml
index 05d34e0539..e2f4bc0edf 100644
--- a/openstack-manila-csi/templates/controllerplugin-service.yaml
+++ b/openstack-manila-csi/templates/controllerplugin-service.yaml
@@ -2,6 +2,7 @@ kind: Service
 apiVersion: v1
 metadata:
   name: {{ include "openstack-manila-csi.controllerplugin.fullname" . }}
+  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "openstack-manila-csi.controllerplugin.labels" .  | nindent 4 }}
 spec:
diff --git a/openstack-manila-csi/templates/controllerplugin-serviceaccount.yaml b/openstack-manila-csi/templates/controllerplugin-serviceaccount.yaml
index c976b9160b..3190363f39 100644
--- a/openstack-manila-csi/templates/controllerplugin-serviceaccount.yaml
+++ b/openstack-manila-csi/templates/controllerplugin-serviceaccount.yaml
@@ -2,5 +2,6 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: {{ include "openstack-manila-csi.serviceAccountName.controllerplugin" . }}
+  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "openstack-manila-csi.controllerplugin.labels" .  | nindent 4 }}
diff --git a/openstack-manila-csi/templates/controllerplugin-statefulset.yaml b/openstack-manila-csi/templates/controllerplugin-statefulset.yaml
index e9263739e4..10bf168417 100644
--- a/openstack-manila-csi/templates/controllerplugin-statefulset.yaml
+++ b/openstack-manila-csi/templates/controllerplugin-statefulset.yaml
@@ -2,6 +2,7 @@ kind: StatefulSet
 apiVersion: apps/v1
 metadata:
   name: {{ include "openstack-manila-csi.controllerplugin.fullname" . }}
+  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "openstack-manila-csi.controllerplugin.labels" .  | nindent 4 }}
 spec:
diff --git a/openstack-manila-csi/templates/nodeplugin-daemonset.yaml b/openstack-manila-csi/templates/nodeplugin-daemonset.yaml
index aa173cc53f..58e2944abe 100644
--- a/openstack-manila-csi/templates/nodeplugin-daemonset.yaml
+++ b/openstack-manila-csi/templates/nodeplugin-daemonset.yaml
@@ -2,6 +2,7 @@ kind: DaemonSet
 apiVersion: apps/v1
 metadata:
   name: {{ include "openstack-manila-csi.nodeplugin.fullname" . }}
+  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "openstack-manila-csi.nodeplugin.labels" .  | nindent 4 }}
 spec:
diff --git a/openstack-manila-csi/templates/nodeplugin-rules-clusterrole.yaml b/openstack-manila-csi/templates/nodeplugin-rules-clusterrole.yaml
index 1f16451a43..bc5c37c5c9 100644
--- a/openstack-manila-csi/templates/nodeplugin-rules-clusterrole.yaml
+++ b/openstack-manila-csi/templates/nodeplugin-rules-clusterrole.yaml
@@ -2,6 +2,7 @@ kind: ClusterRole
 apiVersion: rbac.authorization.k8s.io/v1
 metadata:
   name: {{ include "openstack-manila-csi.nodeplugin.fullname" . }}-rules
+  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "openstack-manila-csi.nodeplugin.labels" .  | nindent 4 }}
     rbac.manila.csi.openstack.org/aggregate-to-nodeplugin-{{ include "openstack-manila-csi.name" . }}: "true"
diff --git a/openstack-manila-csi/templates/nodeplugin-serviceaccount.yaml b/openstack-manila-csi/templates/nodeplugin-serviceaccount.yaml
index 1c7eaab7f8..1a2166f330 100644
--- a/openstack-manila-csi/templates/nodeplugin-serviceaccount.yaml
+++ b/openstack-manila-csi/templates/nodeplugin-serviceaccount.yaml
@@ -2,5 +2,6 @@ apiVersion: v1
 kind: ServiceAccount
 metadata:
   name: {{ include "openstack-manila-csi.serviceAccountName.nodeplugin" . }}
+  namespace: {{ .Release.Namespace }}
   labels:
     {{- include "openstack-manila-csi.nodeplugin.labels" .  | nindent 4 }}
diff --git a/openstack-manila-csi/templates/runtimeconfig-cm.yaml b/openstack-manila-csi/templates/runtimeconfig-cm.yaml
index d739ce1640..59c1a06572 100644
--- a/openstack-manila-csi/templates/runtimeconfig-cm.yaml
+++ b/openstack-manila-csi/templates/runtimeconfig-cm.yaml
@@ -3,6 +3,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: manila-csi-runtimeconf-cm
+  namespace: {{ .Release.Namespace }}
 data:
   runtimeconfig.json: |-
 {{ .Values.csimanila.runtimeConfig.jsonData | indent 4 }}
